packages/web/src/js/FileUploader.ts

```diff
@@ -0,0 +1,261 @@
+  getAttachmentElement(file: File, id: string): HTMLElement | null {
+    const dataContainer = new DataTransfer();
+    const attachmentInputElement = document.createElement('input');
+    const snippetTemplate = SelectorEngine.findOne(TEMPLATE_ELEMENT_SELECTOR, this.element) as HTMLTemplateElement;
+    const snippet = snippetTemplate?.content.cloneNode(true) as HTMLElement;
+
+    if (!snippet) {
+      return null;
+    }
+
+    attachmentInputElement.setAttribute('type', 'file');
+    attachmentInputElement.setAttribute('name', `${this.inputName}[]`);
+    attachmentInputElement.setAttribute('hidden', '');
+    dataContainer.items.add(file);
+    attachmentInputElement.files = dataContainer.files;
+
+    const item = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="item"]`);
+    const itemName = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="name"]`);
+    const itemButton = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="button"]`);
+
+    item!.appendChild(attachmentInputElement);
+    item!.setAttribute('id', id);
+    itemName!.innerHTML = file.name;
+    itemButton!.setAttribute(DATA_DISMISS_ATTRIBUTE, id);
+    item!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+    itemName!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+    itemButton!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+
+    return snippet;
+  }
+
+  createAttachmentEvents(id: string) {
+    const currentItem = SelectorEngine.findOne(`[${DATA_DISMISS_ATTRIBUTE}="${id}"]`);
+    currentItem &&
+      EventHandler.on(currentItem, 'click', (event: Event) => {
+        event.preventDefault();
+        const target = (event.target as HTMLElement)?.dataset?.dismiss;
+        if (target) {
+          this.removeFromQueue(target);
+        }
+      });
+  }
+
+  appendToList(file: File) {
+    if (!this.listElement) {
+      if (process.env.NODE_ENV === 'development') {
+        console.warn('Unfortunately, there is no list element created for inserting files to upload ⚠️');
+      }
+
+      return;
+    }
+
+    if (!this.isMultiple) {
+      this.clearFileQueue();
+    }
+
+    const id = this.getUpdatedFileName(file.name);
+    const attachment = this.getAttachmentElement(file, id);
+
+    if (!attachment) {
+      if (process.env.NODE_ENV === 'development') {
+        console.warn('Cannot find element with template to render ⚠️');
+      }
+
+      return;
+    }
+
+    // Appending an item into queue
+    this.fileQueue.set(id, file);
+
+    // Appending node into list
+    this.listElement.appendChild(attachment);
+
+    // Creating an event for the delete item button
+    this.createAttachmentEvents(id);
+
+    // Reset drag state
+    this.dragReset();
+  }
+
+  addToQueue(file: File) {
+    const checkedFileSize = this.checkAllowedFileSize(file);
+    const checkedDuplicity = this.checkFileQueueDuplicity(file);
+    const run = this.fileQueue.size === 0 ? true : this.fileQueue.size >= 1 && (!checkedDuplicity || !checkedFileSize);
+
+    if (run) {
```

Comment:
I suggest reordering this block:
```js
if (!isFileSizeAllowed) {
  console.warn('The file size limit has been exceeded ⚠️');
  return;
} else if (!isFileQueueUnique) {
  console.warn('This file already exists in the queue ⚠️');
  return;
// is this even possible?
} else if (this.fileQueue.size < 0) {
  console.warn('There was a undefined problem');
  return;
}

this.appendToList(file);
```

Or even better you can skip my suggestion about renaming the check function and make the functions to `throw Error` with the propper message. Than you can use `try/catch` block in this method:
```js
try {
  this.checkAllowedFileSize();
  this.checkDuplicity();
  this.checkSomethingUnhandled()
  this.appendToList();
} catch (error) {
  console.log(error)
}
```

Reply:
And is there a problem with the condition I wrote?

@literat  There are always hundreds of ways to write code in JS, so please let me know if there is a problem somewhere that needs to be fixed. Thank you

Reply:
The problem is the readability and understandability of your code. The functionality and that it just works is not the only benchmark of great code.

You are trying to run the code first and then log all that is wrong. But the logic goes in the opposite direction. First, ensure that all is safe, otherwise log the errors and end the logic.

Secondly, using `try/catch/finally` and `throw Error` is way more effective and readable than using `if/else`, definitely in this scenario. The message itself should be bound to the checker method and its scope. So the `addToQueue` is only responsible for running the checks and appending to the list. `console.log` can be used once.

Reply:
fixed